tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -197,6 +197,42 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "support-list",
 	},
 
+	/** Firmware layout for the CPE210 V3 */
+	{
+		.id     = "CPE210V3",
+		.vendor = "CPE210(TP-LINK|UN|N300-2|00000000):3.0\r\n",
+		.support_list =
+			"SupportList:\r\n"
+			"CPE210(TP-LINK|EU|N300-2|45550000):3.0\r\n"
+			"CPE210(TP-LINK|UN|N300-2|00000000):3.0\r\n"
+			"CPE210(TP-LINK|UN|N300-2):3.0\r\n"
+			"CPE210(TP-LINK|EU|N300-2):3.0\r\n",
+		.support_trail = '\xff',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"partition-table", 0x20000, 0x02000},
+			{"default-mac", 0x30000, 0x00020},
+			{"product-info", 0x31100, 0x00100},
+			{"device-info", 0x31400, 0x00400},
+			{"signature", 0x32000, 0x00400},
+			{"device-id", 0x33000, 0x00100},
+			{"os-image", 0x40000, 0x190000},
+			{"file-system", 0x1d0000, 0x5E0000},
```

Comment:
Could you try if the device boots dynamic partitioning in place for kernel-image / filesystem? See this commit how to switch to dynamic partitioning: https://github.com/openwrt/openwrt/commit/7c78be1b747eb0c8d64da67deb3a8aec75bd7b9c

Reply:
I will test the change - as i know i had this already running for testing. 

Do i need to add MTDPARTS into the Device definition or will the current setting in the DTS File be okay ? (there is a firmware partition)

Reply:
Hi, 

it works with the current commit.